contributing-tips.rst

```diff
@@ -0,0 +1,142 @@
+Tips for Making Good Contributions
+====================================
+
+.. _small-pr:
+
+Smallest meaningful PR
+------------------------
+
+A PR should normally address one issue. This makes it easier to review, easier to deploy, and easier to roll back in case of a problem. Additionally, the smaller the PR, the less likely it is to create a merge issue.
+
+The exception is when several issues are closely related or can reasonably be worked on together.  In this case, it should be clear by looking at the conversation on the Github issues that the items are related and will be worked on together.
+
+
+.. _descriptive-pr-names:
+
+Descriptive PR names
+----------------------
+
+A PR title should answer the question, "What does this Pull Request do?"
+
+Good PR titles:
+
+- adds a video directive
+- makes navigation buttons responsive
+- swaps placement of nav buttons and file an issue note
+
+Bad PR titles:
+
+- fix issue
+- fix #123
+- collect
+
+.. _small-commits:
+
+Small Atomic Commits
+-----------------------
+
+When working locally, commit often. Don't wait until you have 100 lines of changes across multiple files.
+
+- If you need to copy or move a large section or file, commit that change before also editing it. 
+- If you have to create a new template file based on an existing template file, copy the file in one commit and then work on the changes. This makes it easier to know what you actually did.
+- If writing a new doc, commits after each new section are a good idea.
+
+Commit messages should answer the question, "What does this commit do?"
+
+Small, well-named commits will help you keep track of your own work and make rollbacks and other changes easier to deal with.
+
+.. _discuss-issues:
+
+Discuss issues before working
+--------------------------------
+
+Take the time to clarify the needs and scope of an issue before committing to work on it. Especially for coding tasks, make sure you state your understanding and your plan before working. 
+
+If you have a question, ask. Don't guess.
+
+.. note::
+
+  Many new contributors don't ask questions because they are worried about appearing under informed. Please set this worry aside.
+
+  **You will never be judged harshly for asking clarifying questions or for seeking more information.** 
+
+.. _if-you-get-stuck:
+
+If you get stuck while working
+--------------------------------
+
+- Ask for help in the issue comments. Maybe you can get back on track and complete the issue.
+
+  - Asking questions is always better than guessing.
+
+- Submit a ``[WIP]`` (work in progress) pull request.
```

Comment:
Mentioning these PR titles should have *[WIP]* written before the heading should clear this more

Reply:
yes, i need to make that more clear somewhere